src/models/base.js

```diff
@@ -36,19 +39,21 @@ class base {
         data.private_token = this.token;
 
         return new Promise((resolve, reject) => {
-            request.post(`${this.url}${path}`, {
-                json: true,
-                body: data,
-                insecure: this._insecure,
-                proxy: this._proxy,
-                resolveWithFullResponse: true,
-                headers: {
-                    'PRIVATE-TOKEN': this.token
-                }
-            }).then(response => {
-                if (this.config.get('_createDump')) this.setDump(response, key);
-                resolve(response);
-            }).catch(e => reject(e));
+            throttle(() => {
+              request.post(`${this.url}${path}`, {
+                  json: true,
+                  body: data,
+                  insecure: this._insecure,
+                  proxy: this._proxy,
+                  resolveWithFullResponse: true,
+                  headers: {
+                      'PRIVATE-TOKEN': this.token
+                  }
+              }).then(response => {
+                  if (this.config.get('_createDump')) this.setDump(response, key);
+                  resolve(response);
+              }).catch(e => reject(e));
+            })
```

Comment:
Let's move throttle() on line up to minimize the diff
```suggestion
return new Promise((resolve, reject) => throttle(() => {
            request.post(`${this.url}${path}`, {
                json: true,
                body: data,
                insecure: this._insecure,
                proxy: this._proxy,
                resolveWithFullResponse: true,
                headers: {
                    'PRIVATE-TOKEN': this.token
                }
            }).then(response => {
                if (this.config.get('_createDump')) this.setDump(response, key);
                resolve(response);
            }).catch(e => reject(e));
        }));
```